### ISSUE TITLE
Extract array-parameter has wrong type error catching

### ISSUE THREAD

**Issue by Krakabek** [Microsoft/TypeScript#26302]
**TypeScript Version:**  3.1.0-dev.201xxxxx

**Search Terms:**
Array spread types
Spread args array
spread generic types

**Code**
function `call` was found on [msdn blog](https://blogs.msdn.microsoft.com/typescript/2018/07/30/announcing-typescript-3-0/#tuples-and-parameters)
I tried to call it with wrong arguments and get an error, but found a case that is not treated as an error, but it should.

```ts
function call<TS extends any[], R>(fn: (...args: TS) => R, ...args: TS): R {
    return fn(...args);
}

function test(a: number[]) {
}

call(test, [1, 2]); // this is ok
call(test, ["foo", "bar"]); // this has type error and is ok
call(test, ["foo", 1]); // this is treated as valid but is not
```

**Expected behavior:**
Both 2nd and 3rd calls should cause error

**Actual behavior:**
Only 2nd call is wrong, adding at least one number to array of strings makes this legit arg for function call

**Playground Link:** 
[Demo](https://www.typescriptlang.org/play/#src=function%20call%3CTS%20extends%20any%5B%5D%2C%20R%3E(fn%3A%20(...args%3A%20TS)%20%3D%3E%20R%2C%20...args%3A%20TS)%3A%20R%20%7B%0D%0A%20%20%20%20return%20fn(...args)%3B%0D%0A%7D%0D%0A%0D%0Afunction%20test(a%3A%20number%5B%5D)%20%7B%0D%0A%7D%0D%0A%0D%0Acall(test%2C%20%5B1%2C%202%5D)%3B%0D%0Acall(test%2C%20%5B%22foo%22%2C%20%22bar%22%5D)%3B%0D%0Acall(test%2C%20%5B%22foo%22%2C%201%5D)%3B)

**Comment by turbo-lover** [comment]
I have expirienced same problem with this code: 
`
function call<TS extends any[], R>(fn: (...args: TS) => R, ...args: TS): R {
    return fn(...args);
}

interface IFoo{
    bar: number;
}

function test(a: IFoo[]) {
}

call(test, [1, 2]);
call(test, [{ bar: 4 }]);
call(test, ["baz", { bar: 4 }]);
`

**Comment by AnyhowStep** [comment]
`--strictFunctionTypes` ?

**Comment by RyanCavanaugh** [comment]
Unfortunately this is a case of parameter bivariance - you'll need to enable `strictFunctionTypes` to correctly have the error flagged
